features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Shared/ApiKeyRestriction.jsx

```diff
@@ -0,0 +1,335 @@
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import IconButton from "@material-ui/core/IconButton";
+import DeleteIcon from "@material-ui/icons/Delete";
+import Tooltip from "@material-ui/core/Tooltip";
+import Grid from "@material-ui/core/Grid";
+import Fab from "@material-ui/core/Fab";
+import AddIcon from "@material-ui/icons/Add";
+import List from "@material-ui/core/List";
+import ListItem from "@material-ui/core/ListItem";
+import ListItemSecondaryAction from "@material-ui/core/ListItemSecondaryAction";
+import ListItemText from "@material-ui/core/ListItemText";
+import Box from "@material-ui/core/Box";
+import Radio from "@material-ui/core/Radio";
+import RadioGroup from "@material-ui/core/RadioGroup";
+import FormControl from "@material-ui/core/FormControl";
+import Button from "@material-ui/core/Button";
+
+// Styles for Grid and Paper elements
+const styles = (theme) => ({
+  FormControl: {
+    "margin-bottom": "8px",
+    width: "100%",
+    padding: "0px 10px",
+  },
+});
+
+/**
+ * Used to display IP restrictions in generate api key UI
+ */
+const apiKeyRestrictions = (props) => {
+  const [invalidIP, setInvalidIP] = useState(false);
+  const [invalidReferer, setInvalidReferer] = useState(false);
+
+  const {
+    classes,
+    newIP,
+    updateNewIp,
+    ipList,
+    updateIpList,
+    restrictSchema,
+    updateRestrictSchema,
+    newReferer,
+    updateNewReferer,
+    refererList,
+    updateRefererList,
+  } = props;
+
+  const onRefererTextUpdate = (e) => {
+    updateNewReferer(e.target.value.trim());
+    if (e.target.value.trim() == "") {
+      setInvalidReferer(false);
+    }
+  };
+
+  const addRefererItem = () => {
+    if (newReferer != null && newReferer != "") {
+      setInvalidReferer(false);
+      refererList.push(newReferer.trim());
+      updateRefererList(refererList);
+      updateNewReferer("");
+    } else {
+      setInvalidReferer(true);
+    }
+  };
+
+  const deleteRefererItem = (refererItem) => {
+    refererList.splice(refererList.indexOf(refererItem), 1);
+    updateRefererList(refererList);
+  };
+
+  const onIpTextUpdate = (e) => {
+    updateNewIp(e.target.value.trim());
+    if (e.target.value.trim() == "") {
+      setInvalidIP(false);
+    }
+  };
+
+  const addIpItem = () => {
+    if (newIP != null && newIP != "") {
+      const ipv4RegExp = new RegExp(
+        [
+          "^((25[0-5]|2[0-4][0-9]|[01]?",
+          "[0-9][0-9]?).){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)",
+          "(/(3[0-2]|2[0-9]|1[0-9]|[0-9]))?$",
+        ].join("")
+      );
+
+      const ipv6RegExp = new RegExp(
+        [
+          "^(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]",
+          "{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:",
+          "[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4})",
+          "{1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|",
+          "([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]",
```

Comment:
We already have [`@hapi`](https://github.com/wso2/carbon-apimgt/blob/master/features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/package.json#L22) for validation requirements , Shall we use its [ip validation function](https://hapi.dev/module/joi/api/?v=17.1.1#stringipoptions) ?

Reply:
fixed